helm-to-operator-codegen-sdk/common/json_to_string.go

```diff
@@ -0,0 +1,393 @@
+/*
+Copyright 2023 The Nephio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/liyue201/gostl/ds/set"
+	"github.com/liyue201/gostl/ds/stack"
+	"github.com/liyue201/gostl/utils/comparator"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type JsonStringConverter struct {
+	globalStructMapping map[string]string // To be set by Calling Intialise (setModStructMapping)
+	globalEnumsSet      set.Set[string]   // To be set by Calling Intialise (setEnums)
+	generatedGoCode     string            // To be set by jsonToGoCode Function
+}
+
+func (obj *JsonStringConverter) checkOpeningAndClosingBraces() bool {
+	s := stack.New[int]()
+	lineCount := 0
+	opCount, closeCount := 0, 0
+	stringDetected := false
+	for i := 0; i < len(obj.generatedGoCode); i++ {
```

Comment:
I recommend using "for ... range obj.generatedGoCode", which will iterate the runes. Accessing a string by index may result in mistakes for Unicode.